add Kubernetes AI Conformance working group (#8515)
kubernetes/community: wg-ai-conformance/charter.md

# WG Kubernetes AI Conformance Charter

This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
the Roles and Organization Management outlined in [sig-governance].

## Scope

The goal of this group is to define a standardized set of capabilities, APIs, and configurations that a Kubernetes cluster must offer to reliably and efficiently run AI/ML workloads. This initiative aims to simplify AI/ML operations on Kubernetes, accelerate adoption, guarantee interoperability and portability for AI workloads, and enable ecosystem growth on an industry-standard foundation.

### In scope

#### Code, Binaries and Services

> - The primary artifact will be the Kubernetes AI Conformance specification and a suite of tests to demonstrate conformance.

Reviewer: @fedebongio I guess, we should change this, right ?

```suggestion
- The primary artifact will be the CNCF Kubernetes AI Conformance specification and a suite of tests to demonstrate conformance.
```

Reviewer 2: (that can happen later)

#### Cross-cutting and Externally Facing Processes

- The Working Group will consider its primary problem-solving objective complete upon the successful definition and initial adoption of a stable (working title for now) "CNCF Kubernetes AI Conformance" specification.

> - Once the foundational conformance is established and widely recognized, the ongoing maintenance and evolution of the conformance would be evaluated, and could ideally transition to an existing or newly formed Special Interest Group (SIG) with a long-term charter, at which point the Working Group would dissolve.

Reviewer 3: I have certain doubts with this paragraph

 The WG definition in https://github.com/kubernetes/community/blob/master/committee-steering/governance/wg-governance.md#is-it-a-working-group-yes-if
  indicates clearly that a WG

> It does not own any code

A suite of tests is code, and there is also references to 

> and could ideally transition to an existing or newly formed Special Interest Group (SIG) with a long-term charter

So, if this code is in kubernetes it has to be owned by a SIG before, as "transition" implies it was owned by someone before, and a WG can not be the owner as indicated before.

The reference to " newly formed Special Interest Group (SIG) "  is tendentious and irrelevant to the work and scope of this WG, I do not think it should be here.

Reviewer 4: i don't think it is reasonable to define the code ownership in advance, as like k8s conformance the individual parts of tests could be spread across various sigs 
I think it should be a goal to define the SIG ownership within the WG once formed

Reviewer 4: V1 will not have any tests residing in k/k, as there will not be any test, rather than a self assessment questionnaire for companies to fill out (described in the [doc](https://docs.google.com/document/d/1hXoSdh9FEs13Yde8DivCYjjXyxa7j4J8erjZPEGWuzc/edit?tab=t.0#heading=h.3ojn0udl2evq)) 
For future version where feasible (and is technically possible) the idea is to follow the principles of k8s conformance as you described it. However, this part is to be defined and worked out by the working group, as some of the parts might not be as easy to test as k8s feature.

So a potential future Setup could be: SIG tests living in k/k, testing owning organization and architecture of that codebase (TBD with sig testing and interested parties) and the tooling to submit tests residing in CNCF like it does for [k8s](https://github.com/cncf/k8s-conformance)

Yes it might help the to be defined SIG to generate tests and also to help the SIG decide where it is going to be hosted.

> - Once the foundational conformance is established and widely recognized, the ongoing maintenance and evolution of the conformance would be evaluated, and could ideally transition to an existing or newly formed Special Interest Group (SIG) with a long-term charter, at which point the Working Group would dissolve.

Reviewer 3: ```suggestion
- Once the foundational conformance is established and widely recognized, the ongoing maintenance and evolution of the conformance would be evaluated, and could ideally transition to a Special Interest Group (SIG) with a long-term charter, at which point the Working Group would dissolve.
```

> #### Code, Binaries and Services
> 
> - The primary artifact will be the (working title for now) "CNCF Kubernetes AI Conformance" specification and a suite of tests to demonstrate conformance.
> 
> #### Cross-cutting and Externally Facing Processes
> 
> - The Working Group will consider its primary problem-solving objective complete upon the successful definition and initial adoption of a stable (working title for now) "CNCF Kubernetes AI Conformance" specification.
> - Once the foundational conformance is established and widely recognized, the ongoing maintenance and evolution of the conformance would be evaluated, and could ideally transition to a Special Interest Group (SIG) with a long-term charter, at which point the Working Group would dissolve.
> 
> ### Out of scope
> 
> This WG is not responsible for maintenance and evolution of the conformance program, which will be run by the CNCF. Some aspects may be owned by existing SIGs (Architecture, Testing) alongside the CNCF, such as Kubernetes specific testing and tooling, and the pre-existing Kubernetes conformance definition and tests.
> 

Reviewer 4: ```suggestion
#### Code, Binaries and Services

- The primary artifact will be the (working title for now) "CNCF Kubernetes AI Conformance" specification and support for the creation of a suite of tests to demonstrate conformance.

#### Cross-cutting and Externally Facing Processes

- The Working Group will consider its primary problem-solving objective complete upon the successful definition and adoption of a stable (working title for now) "CNCF Kubernetes AI Conformance" specification.
- The first Version will not include any tests in the Kubernetes Codebase and will be a self assessment questionnaire
- Once the initial conformance is established and widely recognized, the ongoing maintenance and evolution of the conformance will be evaluated. Including determining a suitable SIG for Code Organization, Ownership and Architecture and support to create the initial suite of tests. 
After the Ownership is clarified and handed over, the WG dissolves.

### Out of scope

This WG is not responsible for maintenance and evolution of the conformance program, which will be run by the CNCF. Some aspects may be owned by existing SIGs (e.g. Architecture, Testing) alongside the CNCF, such as Kubernetes specific definitions, testing and tooling, and the pre-existing Kubernetes conformance definition and tests.

```

cc @aojea  @BenTheElder

# WG Kubernetes AI Conformance Charter

This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
the Roles and Organization Management outlined in [sig-governance].

## Scope

The goal of this group is to define a standardized set of capabilities, APIs, and configurations that a Kubernetes cluster must offer to reliably and efficiently run AI/ML workloads. This initiative aims to simplify AI/ML operations on Kubernetes, accelerate adoption, guarantee interoperability and portability for AI workloads, and enable ecosystem growth on an industry-standard foundation.

### In scope

#### Code, Binaries and Services

> - The primary artifact will be the Kubernetes AI Conformance specification and a suite of tests to demonstrate conformance.

Reviewer 5: This is an open question from the SIG arch meeting.

IIRC there was loose agreement to follow up in the WG to discuss options for how best to implement and host this and present them to the relevant groups (SIG Arch, SIG Testing, CNCF, ....). 

In think this should be clarified, it is undetermined if there will be binary artifacts in the kubernetes organization. They may be in the CNCF, or the workgroup might determine that there's an alternate approach.

#### Cross-cutting and Externally Facing Processes

> - The Working Group will consider its primary problem-solving objective complete upon the successful definition and initial adoption of a stable Kubernetes AI Conformance specification.

Author: (please don't laugh about my response, maybe I need more coffee). 

I suppose you are suggesting to add the word "CNCF" to get "... adoption of a stable CNCF Kubernetes AI Conformance specification...", right? I agree, that said... 

Charter.md was autogenerated using the [generator](https://github.com/kubernetes/community/tree/master/generator), so I'm not sure how to manually changed it without the CI jobs complaining later that the content was not autogenerated. 
@BenTheElder do you know?

> - The Working Group will consider its primary problem-solving objective complete upon the successful definition and initial adoption of a stable Kubernetes AI Conformance specification.

Author: Fixed. I realized what happened, I think Ben added some clarifications to charter.md (in his role as Kubernetes Steering), so I was trying to find where the extra sigs and the different mission came from. I think we are good now.

- Once the foundational conformance is established and widely recognized, the ongoing maintenance and evolution of the conformance would be evaluated, and could ideally transition to an existing or newly formed Special Interest Group (SIG) with a long-term charter, at which point the Working Group would dissolve.

### Out of scope

This WG is not responsible for the long-term maintenance and evolution of the conformance, which would be evaluated and could ideally transition to an existing or newly formed Special Interest Group (SIG).

## Roles and Organization Management

This WG follows adheres to the Roles and Organization Management outlined in [sig-governance]
and opts-in to updates and modifications to [sig-governance].

### Additional responsibilities of Chairs

- The Working Group will have designated Chair(s) responsible for guiding discussions and ensuring progress.
- A note-taker will be assigned for each meeting, and active participation from all contributors will be encouraged.
- Agendas and meeting notes will be publicly accessible.

### Stakeholder SIGs

> - SIG Architecture (Sponsoring SIG)

Reviewer 6: Do we want to call out specific WGs? For example, WG Batch and WG Serving from SIG Apps should be critical.

> - SIG Architecture (Sponsoring SIG)

Author: In my original email (following the official process) I listed many more SIGs that are affected / involved. I was under the impression that in the PR (and how the generator tool works) only the Owning SIG should be listed, but if I'm wrong I'm happy to change it. What I want to avoid is to have a list of 15 SIGs and having to chase approval from all of them as a formality but at the same time as a blocker to get this going ASAP. 

I think we need a live forum + slack channel, etc as soon as possible, the document ended up being too overloaded by comments.

> - SIG Architecture (Sponsoring SIG)

Author: Repeating from another comment for consistency: does everybody agree to list only: SIG Architecture and SIG Testing for now? Please let me know so I can make the change and get this merged asap to start meeting

> - SIG Architecture (Sponsoring SIG)

Author: Limited the SIGs to Architecture and Testing
